bindings/pyroot/pythonizations/python/ROOT/_pythonization/_th1.py

```diff
@@ -23,10 +24,55 @@ def _imul(self, c):
     return self
 
 
-@pythonization('TH1')
+_np_dtype_dict = {
+    "float64": "TH1D",
+    "float32": "TH1F",
+    "int32": "TH1I",
+    "int8": "TH1C",
+    "int16": "TH1S",
+}
+
+
+def FromNumpy(*args, **kwargs):
+    r"""Function to create histogram object from Numpy arrays"""
+    th, npval = _numpy_getter(_np_dtype_dict, *args, **kwargs)
+    th.FillN(npval[0].size, *npval)
+    return th
+
+
+def _GetBin(self, *args):
+    a = _numpy_content(self._Original_GetBin(*args), *args)
+    return a
+
+
+def _GetBinCenter(self, *args):
+    a = _numpy_content(self._Original_GetBinCenter(*args), *args)
+    return a
+
+
+def _GetBinContent(self, *args):
+    a = _numpy_content(self._Original_GetBinContent(*args), *args)
+    return a
+
+
+def _GetBinLowEdge(self, *args):
+    a = _numpy_content(self._Original_GetBinLowEdge(*args), *args)
+    return a
```

Comment:
I don't see the point of these Pythonizations. All the original functions return a scalar, and in this case the `_numpy_context` function just returns the scalar you pass to it.

Can you please elaborate?

Reply:
Hmm, @guitargeek sir I wasn't sure about the implementation as well, aim was to return content in numpy 
Here's the gist I was using for reference : https://gist.github.com/lmoneta/31912af47c65bb7549bdc7c765e75c02

Reply:
Same for the TH2 and TH3 functions

Reply:
The bin contents are stored in the `fArray` data member of the TH1 (here the [header file](https://github.com/root-project/root/blob/master/hist/hist/inc/TH1.h) for reference).

To implement `GetBinContent` in an efficient way for the full array, you would have access this `fArray` member and create a numpy array from it with `frombuffer`. However, I think this member is not public. Maybe we can add a public accessor to `TH1`, or what would you do here @lmoneta?

Reply:
For the bin centers and edges, it's easier. You can access the array of bin edges with public getters from the TAxis instances in the histogram, which you can for example get like `hist.GetXaxis()`. Then you got to study the [TAxis](https://root.cern.ch/doc/master/classTAxis.html) class a bit to understand how to retrieve bin boundaries best in NumPy.

I guess if you have non, uniform binning, you can get the full bin edges array with `hist.GetXaxis().GetXbins().GetArray()`. For uniform bins, you can just create a `np.linspace` based on the limits and the number of bins

Reply:
@guitargeek Sir when I try to use something like hist.GetSumw2() to convert this I get segmentation error, when I try to return the  bin sum of weight square while implementing GetContent()

[Link](https://gist.github.com/lmoneta/31912af47c65bb7549bdc7c765e75c02) to updated gist.